streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -327,6 +331,9 @@ public void register(final TopicPartition partition, final ProcessorStateManager
             throw new IllegalStateException("There is already a changelog registered for " + partition +
                 ", this should not happen: " + changelogs);
         }
+
+        addChangelogsToRestoreConsumer(Collections.singleton(partition));
```

Comment:
The reason I did not do this is that in `consumer`, even if a partition is paused, we would still try to fetch is fetching position in the `poll` call if it was not available. And that's why I have to defer this after we've found the starting position and initialize them with `seek` calls.

So I'm thinking maybe we would just remove the illegal-state check from the `remoreFromRestoreConsumer` call and just log it as a debug?

Reply:
Ah, I didn't realize the consumer would do anything with paused partitions. I'm not sure we want to remove the verification entirely, but we can check any "missing" partitions against the `changelogs` map instead.